ingress/controllers/nginx/README.md

```diff
@@ -91,242 +94,54 @@ kubectl create -f examples/default-backend.yaml
+This controller provides an alternitive to this module using [nginx-module-vts](https://github.com/vozlt/nginx-module-vts) third party module.
+To use this module just provide a ConfigMap with the key `enable-vts-status=true`. The URL is exposed in the port 8080.
+Please check the example `example/rc-default.yaml`
 
-Like before we need to remove the running nginx rc
-```
-kubectl delete rc nginx-ingress-3rdpartycfg
-```
+![nginx-module-vts screenshot](https://cloud.githubusercontent.com/assets/3648408/10876811/77a67b70-8183-11e5-9924-6a6d0c5dc73a.png "screenshot with filter")
 
-Next create a new rc that uses the secret
-```
-kubectl create -f examples/rc-ssl.yaml
-```
+To extract the information in JSON format the module provides a custom URL: `/nginx_status/format/json`
 
-*Note:* this example uses a self signed certificate.
 
-Example output:
-```
-$ curl -v https://172.17.4.99/foo -H 'Host: bar.baz.com' -k
-*   Trying 172.17.4.99...
-* Connected to 172.17.4.99 (172.17.4.99) port 4444 (#0)
-* TLS 1.2 connection using TLS_ECDHE_RSA_WITH_AES_128_GCM_SHA256
-* Server certificate: foo.bar.com
-> GET /foo HTTP/1.1
-> Host: bar.baz.com
-> User-Agent: curl/7.43.0
-> Accept: */*
->
-< HTTP/1.1 200 OK
-< Server: nginx/1.9.8
-< Date: Thu, 17 Dec 2015 14:57:03 GMT
-< Content-Type: text/plain
-< Transfer-Encoding: chunked
-< Connection: keep-alive
-< Vary: Accept-Encoding
-<
-CLIENT VALUES:
-client_address=10.2.84.34
-command=GET
-real path=/foo
-query=nil
-request_version=1.1
-request_uri=http://bar.baz.com:8080/foo
-
-SERVER VALUES:
-server_version=nginx: 1.9.7 - lua: 9019
-
-HEADERS RECEIVED:
-accept=*/*
-connection=close
-host=bar.baz.com
-user-agent=curl/7.43.0
-x-forwarded-for=172.17.4.1
-x-forwarded-host=bar.baz.com
-x-forwarded-server=bar.baz.com
-x-real-ip=172.17.4.1
-BODY:
-* Connection #0 to host 172.17.4.99 left intact
--no body in request-
-```
+## Troubleshooting
 
+Problems encountered during [1.2.0-alpha7 deployment](https://github.com/kubernetes/kubernetes/blob/master/docs/getting-started-guides/docker.md):
+* make setup-files.sh file in hypercube does not provide 10.0.0.1 IP to make-ca-certs, resulting in CA certs that are issued to the external cluster IP address rather then 10.0.0.1 -> this results in nginx-third-party-lb appearing to get stuck at "Utils.go:177 - Waiting for default/default-http-backend" in the docker logs.  Kubernetes will eventually kill the container before nginx-third-party-lb times out with a message indicating that the CA certificate issuer is invalid (wrong ip), to verify this add zeros to the end of initialDelaySeconds and timeoutSeconds and reload the RC, and docker will log this error before kubernetes kills the container.
+  * To fix the above, setup-files.sh must be patched before the cluster is inited (refer to https://github.com/kubernetes/kubernetes/pull/21504)
 
 ## Custom errors
```

Comment:
these make more sense as nested subheadings of the Troubleshooting section (eg: ### Custom, Debug,  errors)